core/src/main/java/org/testcontainers/containers/output/BaseConsumer.java

```diff
@@ -0,0 +1,76 @@
+package org.testcontainers.containers.output;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+@SuppressWarnings("Duplicates")
+public abstract class BaseConsumer<SELF extends BaseConsumer<SELF>> implements Consumer<OutputFrame> {
+    private StringBuilder logString = new StringBuilder();
+    private OutputFrame brokenFrame;
+    private boolean removeColorCodes = true;
+
+    public SELF withRemoveAnsiCodes(boolean removeAnsiCodes) {
+        this.removeColorCodes = removeAnsiCodes;
+        return (SELF) this;
+    }
+
+    public abstract void process(OutputFrame outputFrame);
+
+    @Override
+    public final synchronized void accept(OutputFrame outputFrame) {
```

Comment:
I'm afraid making it `final` breaks the binary compatibility because there might be some strategies based on i.e. `Slf4jLogConsumer` but overriding `accept` (I think I even have one)